spinoso-time/src/time/tzrs/error.rs

```diff
@@ -0,0 +1,205 @@
+        // Allowing matching arms due to documentation
+        #[allow(clippy::match_same_arms)]
+        match error {
+            // These two are generally recoverable within the usable of `spinoso_time`
+            // TzError::DateTimeError(error) => Self::from(error),
+            TzError::ProjectDateTimeError(error) => Self::from(error),
+
+            // The rest will bleed through, but are included here for reference
+            // Occurs when calling system clock
+            TzError::SystemTimeError(_) => Self::UnknownTzError(error),
+            // Occurs during parsing of TZif files
+            TzError::TzFileError(_) => Self::UnknownTzError(error),
+            // Occurs during parsing of TZif files (POSIX string parsing)
+            TzError::TzStringError(_) => Self::UnknownTzError(error),
+            // Occurs during int conversion (e.g. i64 => i32)
+            TzError::OutOfRangeError(_) => Self::UnknownTzError(error),
+            // Occurs during creation of TimeZoneRef
+            TzError::LocalTimeTypeError(_) => Self::UnknownTzError(error),
+            // Occurs during creation of TimeZoneRef
+            TzError::TransitionRuleError(_) => Self::UnknownTzError(error),
+            // Occurs during creation of TimeZoneRef
+            TzError::TimeZoneError(_) => Self::UnknownTzError(error),
+            // Wrapped by ProjectDateTimeError
+            TzError::FindLocalTimeTypeError(_) => Self::UnknownTzError(error),
+            // Never explicitly returned
+            TzError::IoError(_) => Self::UnknownTzError(error),
+            // Never explicity returned
+            TzError::Utf8Error(_) => Self::UnknownTzError(error),
+            // Never explicitly returned
+            TzError::TryFromSliceError(_) => Self::UnknownTzError(error),
+            // TzError is non_exhaustive, so the rest are caught as Unknown
+            _ => Self::Unknown,
+        }
+    }
+}
+
+impl From<TzStringError> for TimeError {
+    fn from(err: TzStringError) -> Self {
+        Self::TzStringError(err)
+    }
+}
+
+impl From<TzOutOfRangeError> for TimeError {
+    fn from(err: TzOutOfRangeError) -> Self {
+        Self::TzOutOfRangeError(err)
+    }
+}
+
+impl From<IntOverflowError> for TimeError {
+    fn from(error: IntOverflowError) -> Self {
+        Self::IntOverflowError(error)
+    }
+}
+
+#[allow(clippy::module_name_repetitions)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct TzStringError;
+
+impl fmt::Display for TzStringError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
+        write!(
+            f,
+            "+HH:MM\", \"-HH:MM\", \"UTC\" or \"A\"..\"I\",\"K\"..\"Z\" expected for utc_offset"
+        )
+    }
+}
+impl error::Error for TzStringError {}
+
+impl From<Utf8Error> for TzStringError {
+    fn from(_: Utf8Error) -> Self {
+        TzStringError
+    }
+}
+
+#[allow(clippy::module_name_repetitions)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
```

Comment:
```suggestion
#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq, PartialOrd, Ord)]
```

https://rust-lang.github.io/api-guidelines/interoperability.html#types-eagerly-implement-common-traits-c-common-traits